modules/aggregation-layers/src/utils/gpu-grid-aggregation/gpu-grid-aggregator.js

```diff
@@ -201,6 +214,46 @@ export default class GPUGridAggregator {
     return this.runAggregationOnCPU(aggregationParams);
   }
 
+  // Reads aggregation data into JS Array object
+  // For WebGL1, data is available in JS Array objects already.
+  // For WebGL2, data is read from Buffer objects and cached for subsequent queries.
+  /* eslint-disable max-depth */
+  getData(weights = null) {
+    weights = weights && (Array.isArray(weights) ? weights : [weights]);
+    const data = {};
+    const results = this.state.results;
+    const ARRAY_BUFFER_MAP = {
```

Comment:
- Define constants outside
- Any reason you need a map instead of array?

Reply:
Moved const def. outside. Using map , as need both key and value strings.